qiita_db/analysis.py

```diff
@@ -241,9 +241,11 @@ def dropped_samples(self):
         for biom, filepath in viewitems(bioms):
             table = load_table(filepath)
             # remove the samples from the sets as they are found in the table
-            proc_data_id = table.metadata()[0]['Processed_id']
+            proc_data_ids = set(sample['Processed_id']
+                                for sample in table.metadata())
             ids = set(table.ids())
-            all_samples[proc_data_id] = all_samples[proc_data_id] - ids
+            for proc_data_id in proc_data_ids:
```

Comment:
you don't need to go to the command line. On the setUp for the test you can use all the `*.create` functions that you need to update all the data.